### ISSUE TITLE
Publish sets Chmod to 600 on server

### ISSUE THREAD

**Issue by dhornbein** [laktek/punch-sftp-publisher#2]
When I use Punch P, everything works, BUT all the files show up on my server with CHMOD 600.

So when I visit the generated site online it's forbidden!

Config:

```
{
    "template_dir": "templates",
    "content_dir": "contents",
    "output_dir": "public",
    "server": {
        "port": 9009,
        "generate_interval": 10000,
        "serving_only": false
    },
    "bundles": {
        "/scripts/all.js" : [
            "/scripts/vendor/jquery.js",
            "/scripts/vendor/zepto.js",
            "/scripts/foundation.js",
            "/scripts/plugins.js", 
            "/scripts/script.js" 
        ],
        "/scripts/foundation.js" : [ 
            "/scripts/foundation/foundation.js",
            "/scripts/foundation/foundation.alerts.js",
            "/scripts/foundation/foundation.clearing.js",
            "/scripts/foundation/foundation.cookie.js",
            "/scripts/foundation/foundation.dropdown.js",
            "/scripts/foundation/foundation.forms.js",
            "/scripts/foundation/foundation.joyride.js",
            "/scripts/foundation/foundation.magellan.js",
            "/scripts/foundation/foundation.orbit.js",
            "/scripts/foundation/foundation.placeholder.js",
            "/scripts/foundation/foundation.reveal.js",
            "/scripts/foundation/foundation.section.js",
            "/scripts/foundation/foundation.tooltips.js",
            "/scripts/foundation/foundation.topbar.js" 
        ],
        "/css/all.min.css": [
            "/css/foundation.css",
            "/css/layout.css",
            "/css/theme.css",
            "/css/modules/*.css",
            "/css/state.css",
            "/css/prototype.css"
        ]   
    },
    "plugins": {

        "compilers": {
            ".css": "punch-sass-compiler" 
        },
        "publishers": {
            "sftp": "punch-sftp-publisher" 
        }

    },
    "generator": {
        "skip_paths": ["/scss"]
    },
    "publish" : {
        "strategy" : "sftp",
        "options" : {
                "host" : "dev.snip.com",
                "username" : "snip",
                "port" : "22",
                "private_key" : "/home/snip/.ssh/id_rsa",
                "upload_path" : "/snip/"
        }
    }
}
```

`ls -la` on server:

```
drwxr-xr-x 6 snip snip 4096 May  2 19:06 .
drwxrwxr-x 9 root     snip 4096 May  2 18:57 ..
-rw------- 1 snip snip 1011 May  2 19:06 apple-touch-icon.png
drwxr-xr-x 2 snip snip 4096 May  2 19:06 css
-rw------- 1 snip snip 1406 May  2 19:06 favicon.ico
drwxr-xr-x 2 snip snip 4096 May  2 19:06 font
-rw------- 1 snip snip  275 May  2 19:06 humans.txt
drwxr-xr-x 4 snip snip 4096 May  2 19:06 images
-rw------- 1 snip snip    4 May  2 19:06 index.html
drwxr-xr-x 4 snip snip 4096 May  2 19:06 scripts
-rw------- 1 snip snip    4 May  2 19:06 sell.html
-rw------- 1 snip snip    4 May  2 19:06 styleguide.html
```

**Comment by laktek** [comment]
It seems there are few issues with the execution. I'm reopening this.

**Comment by laktek** [comment]
OK, this issue should be fixed from version 1.0.4 or above.
